src/session-pool.ts

```diff
@@ -644,48 +599,24 @@ export class SessionPool extends EventEmitter implements SessionPoolInterface {
     }
     session.lastError = undefined;
 
-    // Release it into the pool as a read-only session in the following cases:
-    // 1. There are more waiters than there are sessions available. Releasing it
-    //    into the pool will ensure that a waiter will be unblocked as soon as
-    //    possible.
-    // 2. The user has not set a write fraction, but this session has been used
-    //    as a read/write session. This is an indication that the application
-    //    needs read/write sessions, and the pool should try to keep that number
-    //    of read/write sessions dynamically.
-    // 3. The user has set a write fraction and that fraction has been reached.
-    const shouldBeWrite =
-      (session.type === types.ReadWrite && this.options.writes === 0) ||
-      (this.options.writes! > 0 &&
-        this.currentWriteFraction < this.options.writes!);
-    if (this.totalWaiters > this.available || !shouldBeWrite) {
-      session.type = types.ReadOnly;
-      this._release(session);
-      return;
-    }
-
     // Delete the trace associated with this session to mark the session as checked
     // back into the pool. This will prevent the session to be marked as leaked if
     // the pool is closed while the session is being prepared.
     this._traces.delete(session.id);
-    this._pendingPrepare++;
```

Comment:
@ko3a4ok This was the only place where `_pendingPrepare` flag was getting updated , now since we have removed this reference , we can remove the property `_pendingPrepare` . Also we should mark `pendingPrepare()` method as deprecated.

Reply:
done.